Library/Homebrew/test/caveats_spec.rb

```diff
@@ -26,4 +26,35 @@ def caveats
       expect(described_class.new(f)).not_to be_empty
     end
   end
+
+  describe "#caveats" do
+    context "when f.plist is not nil" do
+      it "prints plist startup information when f.plist_startup is not nil" do
+        allow(f).to receive(:plist).and_return(true)
+        allow(f).to receive(:plist_startup).and_return(true)
```

Comment:
It would be preferable to use a `formula do` and test this on an actual formula rather than mock these. This also applies to tests below.